kafkaesk/app.py

```diff
@@ -236,22 +250,35 @@ async def __run(self) -> None:
                         partition=record.partition,
                         group_id=self._subscription.group,
                     ).inc()
-                except UnhandledMessage:
-                    # how should we handle this? Right now, fail hard
+                except UnhandledMessage as err:
                     logger.warning(f"Could not process msg: {record}", exc_info=True)
-                    # We increase the error counter
                     CONSUMED_MESSAGES.labels(
                         stream_id=record.topic,
                         partition=record.partition,
                         error="UnhandledMessage",
                         group_id=self._subscription.group,
                     ).inc()
+                    try:
+                        await retry_policy(record=record, error=err)
+                    except UnhandledMessage:
+                        # Do not allow this exception up the stack if it is re-raised
+                        ...
+                except Exception as err:
+                    CONSUMED_MESSAGES.labels(
+                        stream_id=record.topic,
+                        partition=record.partition,
+                        error=err.__class__.__name__,
+                        group_id=self._subscription.group,
+                    ).inc()
+                    await retry_policy(record=record, error=err)
 
                 finally:
                     context.span.finish()
                     context.close()
                     await self.emit("message", record=record)
         finally:
+            # Shutdown the retry policy
+            await retry_policy.finalize()
             try:
                 await self._consumer.commit()
```

Comment:
@tysondenio one question, if the retry policy implementation method `should_requeue` returns true, should be commit anyway? , because requeueing will happen automatically if we dont commit, isnt?